TileDB-Inc/TileDB-Cloud-R
# Flesh out HTTP-retry feature

OpenAPI has auto-generated an HTTP-retry mechanism -- but only on a specified list of error codes, which defaults to the empty list.

On this PR we:

* Apply the logic on a specified list of error codes (408, 502, 503, 504)
* Include the retries in our optional HTTP-debug output

Testing:

* In R `Sys.setenv(TILEDB_CLOUD_R_HTTP_DEBUG='true')` as documented in a vignette update on this PR
* Make some API calls from this package
* Force a server-side 503 (outside the scope of this PR)
* Observe in the displayed output
```
...
 $ handle     :Class 'curl_handle' <externalptr>
 - attr(*, "class")= chr "response"
================================================================
RETRY 1 OF 3 : SLEEP SECONDS 2.038478
================================================================ REQUEST
METHOD: POST
QUERY_PARAMS:
...
 $ handle     :Class 'curl_handle' <externalptr>
 - attr(*, "class")= chr "response"
================================================================
RETRY 2 OF 3 : SLEEP SECONDS 4.554812
================================================================ REQUEST
METHOD: POST
QUERY_PARAMS:
...
 $ handle     :Class 'curl_handle' <externalptr>
 - attr(*, "class")= chr "response"
================================================================
RETRY 3 OF 3 : SLEEP SECONDS 8.777022
================================================================ REQUEST
METHOD: POST
QUERY_PARAMS:
...
```

Also observe that successful requests still complete quickly and correctly.
